### ISSUE TITLE
Always bind `*browser*`

### ISSUE THREAD

**Issue by Ambrevar** [atlas-engineer/nyxt#2670]
In https://github.com/atlas-engineer/nyxt/pull/2666#discussion_r1031497582 we discussed the problem of having to protect against NIL `*browser*` all over the place.  Then having to extract some default value, which duplicates it since it's already in the slot initforms.

Suggestion: always bind `*browser*` to at least `(make-instance 'browser)`.

I believe this should do the right thing.  Thoughts?

**Comment by aadcg** [comment]
Agreed.

**Comment by aadcg** [comment]
Sorry for bikesheding this topic, but are there cases when we really must use *browser*?

There are cases when we call it, but we should define a method in the `browser` class. For instance, see  `buffer-list', `buffers-get', `buffers-set' or `buffers-delete'.

**Comment by Ambrevar** [comment]
Sorry I didn't understand.  Can you rephrase / give an example?

**Comment by jmercouris** [comment]
If `*browser*` is unbound, then nothing will work, which is OK with me. If it is unbound, nothing *should* work.
